### ISSUE TITLE
net/wireguard: Firewall ruleset not created

### ISSUE THREAD

**Issue by kevemueller** [opnsense/plugins#1548]
On a rather fresh install of OPNsense 19.7.5_5-amd64 adding the plugin wireguard (v1.1) differs in behaviour from older releases. I am comparing against OPNsense 19.7.4_1-amd64 with the same plugin version (v1.1).

On this setup adding the plugin does not create a new ruleset "Wireguard" under Firewall/Rules.
The plugin can be configured well and connections are established, only that incoming packets are blocked by the default routes.
Uninstalling/re-installing the plugin did not resolve the issue.

Please help me troubleshoot the issue.

**Comment by kevemueller** [comment]
New requirement is to do an interface assignment. Was apparently not necessary in the previous version.
With interface assigned, everything looks good.

**Comment by bnbartekc** [comment]
Could this be reopened and changed to feature request?
Currently (OPNsense version 21.7.1 and WireGuard plugin version 1.7) after installing wireguard plugin, configuring an instance and starting the service the common wireguard interface (not visible in GUI but present in config.xml) and wireguard firewall rulest do **not** get created. They only appear after interface of already existing instance (which would usually be wg0) is assigned a description. It doesn't even need to be enabled. Afterwards even if interface assignment is removed and wireguard plugin is uninstalled that common interface in config.xml and firewall ruleset persist. 
I think it would be nice to have the wireguard interface in config.xml and wireguard ruleset in firewall GUI (Firewall->Rules) created when plugin is installed and removed when plugin is deleted. This would allow automatic creation of firewall rules via API for Wireguard instances without having to first manually force creation of wireguard interface.
